src/app/models/feedback.py

```diff
@@ -0,0 +1,120 @@
+            feedback_subject: dbFeedback = aliased(dbFeedback, name="feedback_subject")
+            player_name: dbPlayer = aliased(dbPlayer, name="player_name")
+
+            query = select(
+                [
+                    feedback_voter.vote,
+                    feedback_voter.prediction,
+                    feedback_voter.confidence,
+                    player_name.name,
+                    feedback_voter.feedback_text,
+                    feedback_voter.proposed_label,
+                ]
+            )
+            query = query.select_from(player_name)
+            query = query.join(
+                feedback_subject, feedback_subject.subject_id == player_name.id
+            )
+            query = query.join(
+                feedback_voter, feedback_voter.voter_id == player_name.id
+            )
+            query = query.where(
+                text("player_name.name IN :names").bindparams(
+                    names=player_names
+                )  # prevent sql injection
+            )
+            result: Result = await self.session.execute(query)
+            await self.session.commit()
+
+        result_set = result.mappings().all()
+        feedback_responses = [
+            ResponseFeedback(
+                player_name=feedback.name,
+                vote=feedback.vote,
+                prediction=feedback.prediction,
+                confidence=feedback.confidence,
+                feedback_text=feedback.feedback_text,
+                proposed_label=feedback.proposed_label,
+            )
+            for feedback in result_set
+        ]
+
+        # logging.debug(f"Output result: {feedback_responses}")
+
+        return feedback_responses
+
+    async def get_feedback_score(self, player_names: list[str]):
+        async with self.session:
+            # print(f"Player names: {player_names}")
+            feedback_voter: dbFeedback = aliased(dbFeedback, name="feedback_voter")
+            reported_player: dbPlayer = aliased(dbPlayer, name="reported_player")
+
+            # Revised code
+            subquery = (
+                select(
+                    [
+                        func.count(feedback_voter.vote).label("count"),
+                        feedback_voter.voter_id,
+                    ]
+                )
+                .group_by(feedback_voter.voter_id)
+                .subquery()
+            )
+
+            query = select(
+                [
+                    subquery.c.count,
+                    reported_player.possible_ban,
+                    reported_player.confirmed_ban,
+                    reported_player.confirmed_player,
+                ]
+            )
+            query = query.join(
+                reported_player, subquery.c.voter_id == reported_player.id
+            )
+            query = query.where(
+                text("reported_player.name IN :names").bindparams(names=player_names)
+            )  # prevent sql injection
```

Comment:
this should be the query.

note for semantic reasons in the join we set the left of the "==" to the table we are joining to: "dbFeedback" because the default behaviour of a join is a left join.

```sql
feedback_voter: dbPlayer = aliased(dbPlayer, name="feedback_voter")
feedback_subject: dbPlayer = aliased(dbPlayer, name="feedback_subject")

query: Select = select(
	[
		func.count(func.distinct(feedback_subject.id)).label("count"),
		feedback_subject.possible_ban,
		feedback_subject.confirmed_ban,
		feedback_subject.confirmed_player,
		dbFeedback.vote,
	]
)
query = query.select_from(dbFeedback)
query = query.join(
	feedback_voter, dbFeedback.voter_id == feedback_voter.id
)
query = query.join(
	feedback_subject, dbFeedback.subject_id == feedback_subject.id
)
query = query.where(feedback_voter.name.in_(player_names))
query = query.group_by(
	feedback_subject.possible_ban,
	feedback_subject.confirmed_ban,
	feedback_subject.confirmed_player,
	dbFeedback.vote,
)
```

Reply:
this is resolve, however this is the wrong function (get_feedback vs get feedback score)
get_feedback resolution:
https://github.com/Bot-detector/public-api/blob/91bdf391df482e4511ec664118865f178383a9cc/src/app/models/feedback.py#L38-L41

this is for get_feedback_score:
https://github.com/Bot-detector/public-api/blob/91bdf391df482e4511ec664118865f178383a9cc/src/app/models/feedback.py#L89-L92

note the subquery is defined here:
https://github.com/Bot-detector/public-api/blob/91bdf391df482e4511ec664118865f178383a9cc/src/app/models/feedback.py#L70-L79

Reply:
this is the new join for feedback score fixed
[name consistancy and correct join](https://github.com/Bot-detector/public-api/pull/25/commits/8fd4f95f35d41ded9129428567b3b6a34dd49dc4)